Userland/Libraries/LibGfx/DDSLoader.cpp

```diff
@@ -0,0 +1,1064 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Debug.h>
+#include <AK/Endian.h>
+#include <AK/LexicalPath.h>
+#include <AK/MappedFile.h>
+#include <AK/StringBuilder.h>
+#include <LibGfx/DDSLoader.h>
+#include <fcntl.h>
+#include <math.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#ifdef __serenity__
+#    include <serenity.h>
+#endif
+
+namespace Gfx {
+
+class Streamer {
+public:
+    Streamer(const u8* data, size_t size)
+        : m_data_ptr(data)
+        , m_size_remaining(size)
+    {
+    }
+
+    template<typename T>
+    bool read(T& value)
+    {
+        if (m_size_remaining < sizeof(T))
+            return false;
+        value = *((const LittleEndian<T>*)m_data_ptr);
+        m_data_ptr += sizeof(T);
+        m_size_remaining -= sizeof(T);
+        return true;
+    }
+
+    bool drop_bytes(size_t count)
+    {
+        if (m_size_remaining < count)
+            return false;
+        m_data_ptr += count;
+        m_size_remaining -= count;
+        return true;
+    }
+
+    bool at_end() const { return !m_size_remaining; }
+
+    size_t remaining() const { return m_size_remaining; }
+
+private:
+    const u8* m_data_ptr { nullptr };
+    size_t m_size_remaining { 0 };
+};
```

Comment:
any reason for adding this instead of just using AK::InputMemoryStream?

Reply:
I saw the same block of code used in the other image loaders (And didnt know about the `InputMemoryStream`).

Reply:
It does not, but i dont think that's a bad thing, just calling a class called streamer and expecting LE reads back is not very obvious :)

Reply:
Swapped to `InputMemoryStream` now 👍